opencensus/stats/internal/tag_key.cc

```diff
@@ -0,0 +1,76 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "opencensus/stats/tag_key.h"
+
+#include <cstdint>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "absl/strings/string_view.h"
+#include "absl/synchronization/mutex.h"
+
+namespace opencensus {
+namespace stats {
+
+class TagKeyRegistry final {
+ public:
+  static TagKeyRegistry* Get() {
+    static TagKeyRegistry* global_tag_key_registry = new TagKeyRegistry();
+    return global_tag_key_registry;
+  }
+
+  TagKey Register(absl::string_view name);
+
+  // Return string_view rather than a const ref so that returned values will not
```

Comment:
But it returns a constref?

Reply:
Removed comment. I went back on that because most of the uses required a std::string and the string_view interface thus caused a lot of copies. This should not be a problem in normal use where tag keys are initialized during startup.